### ISSUE TITLE
Nesting inside a positioned element causes incorrect corrective values

### ISSUE THREAD

**Issue by endash** [gss/engine#165]
Given two divs, one at the top level and another nested in a div, with the following rules and structure:

    <style type="text/gss">
      #one[top] == 0;
      #two[top] == #one[top];
      #one[height] == 200;
      #two[height] == 100;
      #one[width] == ::window[width];
      #two[width] == ::window[width];
    </style>
    <body>
        <div id="one"></div>
        <div id="container">
          <div id="two"></div>
        </div>
    </body>

Div two will only be positioned properly if container is unpositioned (static). With a relative or absolute positioning Div two will have a `top` value of -200, when in fact 0 would be correct. This varies as you would expect with the height of Div one.

It's possible I'm misunderstanding the intended behaviour, but if correct alignment across nesting isn't intended then it doesn't seem like there should be any effect applied at all.

**Comment by endash** [comment]
Just a hunch, but since adding the `top` constraint causes the two previously statically positioned divs to become absolutely positioned, I'm guessing the calculation for how to adjust Div `two`'s position across the nesting is based on the positions of the Divs prior to the absolute positioning being applied.

**Comment by Inviz** [comment]
Indeed, this happens. 

It does measurement & styling in two separate steps. So when it measures things, it doesnt know yet that something will be taken out of the page flow.

Then it validates those values. Quite counter-intuitively it does not do a validation pass if nothing in the document is listening intrinsic values. This is an incorrect assumption - I'll fix it. 

Using something like the following doesnt let the first frame to use stale positioning information by triggering another remeasurement pass.

      a == #two[computed-width]; // make something trigger validation pass
      #container {
        position: relative;
      }

**Comment by Inviz** [comment]
Though generally, position is completely ignored and top/left value should be considered absolute to the document. Adding position: relative on the element helps to maintain multiple global rectangles in the document (there's a mini-guide to this in docs)

**Comment by endash** [comment]
I was exploring how well GSS can coexist with plain jane HTML/CSS, as I'm intending to use it as the basis for an extension to Ember.js playing nicely is important. I don't know how often this precise scenario would actually happen, but it's certainly not out of the realm of possibility.

**Comment by endash** [comment]
I was exploring how well GSS can coexist with plain jane HTML/CSS, as I'm intending to use it as the basis for an extension to Ember.js and playing nicely is important. I don't know how often this precise scenario would actually happen, but it's certainly not out of the realm of possibility.

**Comment by Inviz** [comment]
Sure. Another possible suggestion could be to set your main blocks be positioned to `absolute` value from the start. To take them out of the flow by the time first measurements happen.

You can also bind to intrinsic-y/computed-y of elements which should not take the anchor element out of the flow. But it's a bit undertested area at the moment (there's at least one bug issued)

**Comment by endash** [comment]
And with a JS app there will be changes that should flush out any stale values. I'm not gonna worry about it until it proves itself to be a real issue for my project. Setting absolute from the start with CSS is a fix, but it's also the sort of arcane implementation detail I need to shield from end users. Complete transparency is the goal. This is, to be fair, a contrived example.

**Comment by Inviz** [comment]
I agree with you, we'll ensure another measurement pass when switching `position` style of elements, or something like that.
